cmd/mdatagen/internal/templates/config_test.go.tmpl

```diff
@@ -73,7 +75,19 @@ func loadMetricsBuilderConfig(t *testing.T, name string) MetricsBuilderConfig {
 	sub, err := cm.Sub(name)
 	require.NoError(t, err)
 	cfg := DefaultMetricsBuilderConfig()
-	require.NoError(t, sub.Unmarshal(&cfg))
+	require.NoError(t, sub.Unmarshal(&cfg, confmap.WithIgnoreUnused()))
```

Comment:
Why `WithIgnoreUnused` added here and to events?

Reply:
The `DefaultMetricsBuilderConfig()` function currently defines only the `metrics` and `resources` sections for the `metadata.yml` file. We are now introducing an `events` section at the same level as `metrics`. However, this addition will cause the `Unmarshal` process to fail because it will encounter unused sections.

Reply:
Ok. Potentially we can split the `testdata/config.yaml` I guess.